shard.go

```diff
@@ -75,16 +87,42 @@ func (s *cacheShard) set(key string, hashedKey uint64, entry []byte) error {
 	for {
 		if index, err := s.entries.Push(w); err == nil {
 			s.hashmap[hashedKey] = uint32(index)
-			s.lock.Unlock()
+			if !skipLock {
+				s.lock.Unlock()
+			}
 			return nil
 		}
 		if s.removeOldestEntry(NoSpace) != nil {
-			s.lock.Unlock()
+			if !skipLock {
+				s.lock.Unlock()
+			}
 			return fmt.Errorf("entry is bigger than max shard size")
 		}
 	}
 }
 
+func (s *cacheShard) append(key string, hashedKey uint64, entry []byte) error {
+	s.lock.Lock()
+	var newEntry []byte
+	oldEntry, err := s.get(key, hashedKey, true)
+	if err == nil {
+		newEntry = oldEntry
+	} else if err != ErrEntryNotFound {
+		s.lock.Unlock()
+		return err
+	} else {
+		// Entry not found
+	}
```

Comment:
I suppose this looks a bit more easy to follow, what do you think?
```
if err != nil {
	if err != ErrEntryNotFound {
		s.lock.Unlock()
		return err
	}
} else {
	newEntry = oldEntry
}
```

Reply:
yes changed it